### PR TITLE
[PLGN-619] Anomali Threatstream - Update SDK

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Update SDK to newest version
  - Add unit tests
  - Add API class to connection

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/anomali_threatstream/komand_anomali_threatstream/util/api.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        logger: Logger,
+    ):
+        self.session = Session()
+        self.request = Request()
+        self.logger = logger
+        self.request.url = url
+        self.request.verify = verify
+        self.request.params = params
+
+    def send(self, request):
```
Comment (igorski-r7) [inline plugins/anomali_threatstream/komand_anomali_threatstream/util/api.py:25]:
```suggestion
    def send(self, request: Request) -> Dict[str, Any]:
```

File: plugins/anomali_threatstream/komand_anomali_threatstream/util/api.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    def hide_api_key(string):
+        """
+        ThreatStream queries expose the API key as a URL query parameter. This method hides the API key using a regex that
+        substitutes with the replacement the first instance in string of a substring that matches pattern.
+        The pattern regex matches the api_key URL query parameter. It captures whether or not another parameter
+        follows the api_key value in the group named" "end" with an &. The replacement regex retains the end group.
+        """
+        pattern = r"api_key=([a-zA-Z0-9]+)(?P<end>\&|$)"
+        replacement = r"api_key=********\g<end>"
+        return re.sub(pattern, replacement, string)
```
Comment (igorski-r7) [inline plugins/anomali_threatstream/komand_anomali_threatstream/util/api.py:64]:
```suggestion
    def hide_api_key(input_string: str) -> str:
        """
        ThreatStream queries expose the API key as a URL query parameter. This method hides the API key using a regex that
        substitutes with the replacement the first instance in string of a substring that matches pattern.
        The pattern regex matches the api_key URL query parameter. It captures whether or not another parameter
        follows the api_key value in the group named" "end" with an &. The replacement regex retains the end group.
        """
        pattern = r"api_key=([a-zA-Z0-9]+)(?P<end>\&|$)"
        replacement = r"api_key=********\g<end>"
        return re.sub(pattern, replacement, input_string)
```

File: plugins/anomali_threatstream/plugin.spec.yaml [hunk 3]
```diff
@@ -5,9 +5,15 @@ name: anomali_threatstream
...
-version: 3.1.1
-vendor: rapid7
+version: 3.1.2
+connection_version: 3
+vendor: rapid7_sentry
 support: community
+supported_versions: ["Anomali API API 22-02-2022"]
+sdk:
+  type: slim
+  version: 5
```
Comment (igorski-r7) [inline plugins/anomali_threatstream/plugin.spec.yaml:15]:
```suggestion
  version: 5.4.4
```

File: plugins/anomali_threatstream/plugin.spec.yaml [hunk 4]
```diff
@@ -535,11 +544,13 @@ actions:
         description: Operation status
         type: boolean
         required: false
+        example: true
       reports:
         title: Reports
         description: Reports containing submission details
         type: "[]report"
         required: false
+        example: []
```
Comment (igorski-r7) [inline plugins/anomali_threatstream/plugin.spec.yaml:553]: Could you please fill out that example? 😄

File: plugins/anomali_threatstream/unit_test/expected/submit_file_success.json.exp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/expected/submit_file_success.json.exp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/expected/submit_url_success.json.exp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/expected/submit_url_success.json.exp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/inputs/import_observables_success.json.inp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/inputs/import_observables_success.json.inp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/responses/import_observable_success.json.resp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/responses/import_observable_success.json.resp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/responses/lookup_hash_success.json.resp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/responses/lookup_hash_success.json.resp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/responses/lookup_ip_success.json.resp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/responses/lookup_ip_success.json.resp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/responses/lookup_url_success.json.resp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/responses/lookup_url_success.json.resp:1]: Could you please pretty-print it?

File: plugins/anomali_threatstream/unit_test/responses/submit_file_success.json.resp
Comment (igorski-r7) [inline plugins/anomali_threatstream/unit_test/responses/submit_file_success.json.resp:1]: Could you please pretty-print it?

**Review by rbowden-r7 — APPROVED** [review]

**Review by dlaverty-r7 — APPROVED** [review]

**Pull request merged**